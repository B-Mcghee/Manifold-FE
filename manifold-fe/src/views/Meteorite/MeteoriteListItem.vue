<template>
  <div>
    <base-card class="mx-auto my-2" data-testid="base-card">
      <template v-slot:header>
        <v-tooltip v-if="meteorite.name.length > 13" top>
          <template v-slot:activator="{ on, attrs }">
            <v-toolbar-title v-bind="attrs" v-on="on" data-testid="toolbar">{{
              meteorite.name | formatName
            }}</v-toolbar-title>
          </template>
          <span>{{ meteorite.name }}</span>
        </v-tooltip>
        <v-toolbar-title v-else>{{
          meteorite.name | formatName
        }}</v-toolbar-title>

        <v-spacer></v-spacer>
        <h3 class="text-h6">ID: {{ meteorite.id }}</h3>
      </template>
      <template v-slot:body>
        <v-container>
          <v-layout row>
            <v-flex xs12 class="pl-4 ">
              Mass:
              <span class="blue--text text--darken-3">
                {{ meteorite.mass }}(g)</span
              >
            </v-flex>
            <v-flex xs12 class="pl-4 mt-1">
              Reclat:
              <span class="blue--text text--darken-3"
                >({{ meteorite.reclat }})</span
              >
            </v-flex>
            <v-flex xs12 class="pl-4 mt-1">
              Reclong:
              <span class="blue--text text--darken-3"
                >({{ meteorite.reclong }})</span
              >
            </v-flex>
            <v-flex xs12 class="pl-4 mt-1">
              Nametype:
              <span class="blue--text text--darken-3">{{
                meteorite.nametype
              }}</span>
            </v-flex>
            <v-flex xs12 class="pl-4 mt-1">
              Fall:
              <span class="blue--text text--darken-3">{{
                meteorite.fall
              }}</span>
            </v-flex>
          </v-layout>
        </v-container>
      </template>
      <template v-slot:footer>
        <v-layout block row class="py-3">
          <v-flex block xs12>
            <v-btn
              x-small
              outlined
              text
              width="100%"
              height="48px"
              class="text-subtitle-1"
            >
              <router-link
                :to="{
                  name: 'Recclass',
                  params: { recclass: meteorite.recclass },
                }"
                >Reclass: {{ meteorite.recclass }}</router-link
              >
            </v-btn>
          </v-flex>
          <v-flex xs12>
            <v-btn
              x-small
              outlined
              text
              width="100%"
              height="48px"
              class="text-subtitle-1"
              ><router-link
                :to="{
                  name: 'Year',
                  params: { year: meteorite.year },
                }"
              >
                year:{{ meteorite.year | formatDate }}</router-link
              ></v-btn
            >
          </v-flex>

          <v-flex xs6 class="mt-1 mx-auto ">
            <favorite-icon
              :favorite="JSON.parse(meteorite.id)"
              ref="icon"
            ></favorite-icon>
          </v-flex>
          <v-flex xs6 class="mt-1">
            <compare-icon :meteorite="JSON.parse(meteorite.id)"></compare-icon>
          </v-flex>
        </v-layout>
      </template>
    </base-card>
  </div>
</template>

<script>
import CompareIcon from "@/components/UI/CompareIcon.vue";
import FavoriteIcon from "@/components/UI/FavoriteIcon.vue";
import BaseCard from "@/components/UI/BaseCard.vue";
export default {
  components: { FavoriteIcon, CompareIcon, BaseCard },
  props: {
    meteorite: Object,
  },
  data: () => ({
    favorite: false,
  }),
  methods: {
    resortChild(recclass) {
      console.log(recclass);
      this.$emit("resort", recclass);
    },
    toggle() {},
  },

  filters: {
    formatDate: function(value) {
      if (value) {
        var date = new Date(value);
        var parsedDate = date.getFullYear();
        return parsedDate;
      } else {
        return "N/A";
      }
    },
    formatName: function(value) {
      if (value.length > 15) {
        return value.substring(0, 15);
      }
      return value;
    },
    formatLocation: function(values) {
      if (values) {
        var x = JSON.parse(JSON.stringify(values));
        if (x.geolocation) {
          var geolocation = JSON.parse(JSON.stringify(x.geolocation));
          return geolocation.type + " [" + geolocation.coordinates + "]";
        }
      }
    },
  },
  computed: {
    isFavorite: function() {
      return "";
    },
  },
};
</script>

<style>
a {
  text-decoration: none;
}
</style>
