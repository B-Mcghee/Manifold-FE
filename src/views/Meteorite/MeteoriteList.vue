<template>
  <div>
    <div
      class="  d-flex justify-center text-h5  pb-4 font-weight-bold primary--text text-lg-h4"
    >
      <h1 v-if="!year">{{ title }}</h1>
      <h1 v-else>{{ title }}</h1>
    </div>

    <meteorite-loader :meteorites="loadMeteorites"></meteorite-loader>
  </div>
</template>

<script>
import { mapGetters } from "vuex";
import MeteoriteLoader from "@/components/MeteoriteLoader.vue";
export default {
  props: {
    category: {
      type: Object
    }
  },
  components: { MeteoriteLoader },
  data() {
    return {
      year: false,
      title: "Meteorites"
    };
  },
  methods: {
    allMeteorites() {
      return this.initialMeteorites;
    },
    favoriteList() {
      const array = [];
      this.favorites.forEach(m => {
        this.initialMeteorites.filter(meteorite => {
          if (m == meteorite.id) {
            array.push(meteorite);
          }
        });
      });
      return array;
    },
    comparisonList() {
      const array = [];
      this.meteoriteComparison.forEach(m => {
        this.initialMeteorites.filter(meteorite => {
          if (m == meteorite.id) {
            array.push(meteorite);
          }
        });
      });
      return array;
    },
    propertyList(value) {
      value === "year"
        ? this.alteredTitle(`${this.category.value.substring(0, 4)} /`)
        : this.alteredTitle(`${this.category.value} /`);

      const array = this.initialMeteorites.filter(
        meteorite =>
          meteorite[`${this.category.property}`] == this.category.value
      );
      return array;
    },
    alteredTitle(value) {
      this.title = value + " Meteorites";
    }
  },

  computed: {
    ...mapGetters(["favorites", "initialMeteorites", "meteoriteComparison"]),
    loadMeteorites: function() {
      if (this.category.category) {
        switch (this.category.category) {
          case "Compare":
            this.alteredTitle("Compare");
            return this.comparisonList();
          case "Favorites":
            this.alteredTitle("Favorite");
            return this.favoriteList();
          default:
            break;
        }
      } else if (this.category.property) {
        return this.propertyList(this.category.property);
      }
      this.alteredTitle("All");
      return this.allMeteorites();
    }
  },
  filters: {}
};
</script>

<style></style>
