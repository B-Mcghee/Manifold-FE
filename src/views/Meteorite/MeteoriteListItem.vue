<template>
  <div>
    <base-card class="mx-auto my-2" data-testid="base-card">
      <template v-slot:header>
        <v-tooltip v-if="meteorite.name.length > 13" top>
          <template v-slot:activator="{ on, attrs }">
            <v-toolbar-title v-bind="attrs" v-on="on" data-testid="toolbar">{{
              meteorite.name | formatName
            }}</v-toolbar-title>
          </template>
          <span>{{ meteorite.name }}</span>
        </v-tooltip>
        <v-toolbar-title v-else>{{
          meteorite.name | formatName
        }}</v-toolbar-title>

        <v-spacer></v-spacer>
        <h3 class="text-h6">ID: {{ meteorite.id }}</h3>
      </template>
      <template v-slot:body>
        <v-container>
          <v-layout row>
            <v-flex xs12 class="pl-4 ">
              Mass:
              <span class="blue--text text--darken-3">
                {{ meteorite.mass }}(g)</span
              >
            </v-flex>
            <v-flex xs12 class="pl-4 mt-1">
              Reclat:
              <span class="blue--text text--darken-3"
                >({{ meteorite.reclat }})</span
              >
            </v-flex>
            <v-flex xs12 class="pl-4 mt-1">
              Reclong:
              <span class="blue--text text--darken-3"
                >({{ meteorite.reclong }})</span
              >
            </v-flex>
            <v-flex xs12 class="pl-4 mt-1">
              Nametype:
              <span class="blue--text text--darken-3">{{
                meteorite.nametype
              }}</span>
            </v-flex>
            <v-flex xs12 class="pl-4 mt-1">
              Fall:
              <span class="blue--text text--darken-3">{{
                meteorite.fall
              }}</span>
            </v-flex>
          </v-layout>
        </v-container>
      </template>
      <template v-slot:footer>
        <v-layout block row class="py-3">
          <v-flex block xs12>
            <v-btn
              x-small
              outlined
              text
              width="100%"
              height="48px"
              class="text-subtitle-1 darkPrimary--text"
              :to="{
                name: 'Property',
                params: {
                  property: 'recclass',
                  value: meteorite.recclass,
                },
              }"
            >
              Reclass: {{ meteorite.recclass }}
            </v-btn>
          </v-flex>
          <v-flex xs12>
            <v-btn
              x-small
              outlined
              text
              width="100%"
              height="48px"
              class="text-subtitle-1 darkPrimary--text"
              :to="{
                name: 'Property',
                params: { property: 'year', value: meteorite.year },
              }"
            >
              year:{{ meteorite.year | formatDate }}</v-btn
            >
          </v-flex>

          <v-flex xs6 class="mt-1 mx-auto ">
            <favorite-icon
              @toggleFavorite="toggleFavorite"
              :favorite="isFavorite"
              ref="icon"
            ></favorite-icon>
          </v-flex>
          <v-flex xs6 class="mt-1">
            <compare-icon
              @compareMeteorite="compareMeteorite"
              :compare="toCompare"
            ></compare-icon>
          </v-flex>
        </v-layout>
      </template>
    </base-card>
  </div>
</template>

<script>
import CompareIcon from "@/components/UI/CompareIcon.vue";
import FavoriteIcon from "@/components/UI/FavoriteIcon.vue";
import BaseCard from "@/components/UI/BaseCard.vue";
import { mapGetters, mapActions } from "vuex";

export default {
  components: { FavoriteIcon, CompareIcon, BaseCard },
  props: {
    meteorite: Object,
  },
  methods: {
    ...mapActions([
      "addToFavorites",
      "removeFavorite",
      "addComparison",
      "removeComparison",
    ]),
    checkFavorite(value) {
      return value === JSON.parse(this.meteorite.id);
    },
    toggleFavorite() {
      if (!this.favorites.includes(this.cleanId)) {
        console.log(this.addToFavorites);
        this.addToFavorites(this.cleanId);
      } else {
        this.removeFavorite(this.cleanId);
      }
    },
    compareMeteorite() {
      if (!this.meteoriteComparison.includes(this.cleanId)) {
        this.addComparison(this.cleanId);
      } else {
        this.removeComparison(this.cleanId);
      }
    },
  },

  filters: {
    formatDate: function(value) {
      if (value) {
        var date = new Date(value);
        var parsedDate = date.getFullYear();
        return parsedDate;
      } else {
        return "N/A";
      }
    },
    formatName: function(value) {
      if (value.length > 15) {
        return value.substring(0, 15);
      }
      return value;
    },
    formatLocation: function(values) {
      if (values) {
        var x = JSON.parse(JSON.stringify(values));
        if (x.geolocation) {
          var geolocation = JSON.parse(JSON.stringify(x.geolocation));
          return geolocation.type + " [" + geolocation.coordinates + "]";
        }
      }
    },
  },
  computed: {
    ...mapGetters(["favorites", "initialMeteorites", "meteoriteComparison"]),
    isFavorite() {
      const num = this.favorites.findIndex(this.checkFavorite);
      return num !== -1 ? true : false;
    },
    toCompare() {
      const num = this.meteoriteComparison.findIndex(this.checkFavorite);
      return num !== -1 ? true : false;
    },
    cleanId: function() {
      return JSON.parse(this.meteorite.id);
    },
  },
};
</script>

<style>
a {
  text-decoration: none;
}
</style>
