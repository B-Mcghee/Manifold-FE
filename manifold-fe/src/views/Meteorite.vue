<template>
  <div>
    <div class="  d-flex justify-center text-h3 my-6 font-weight-light">
      Meteorites
    </div>
    <div class="list-container">
      <meteorite-list-item :meteorite="meteorite"></meteorite-list-item>
      <!-- <meteorite-list></meteorite-list> -->
    </div>
  </div>
</template>

<script>
// import MeteoriteList from "../components/MeteoriteList";
// import MeteoriteListItem from "../components/MeteoriteListItem";
import MeteoriteListItem from "../components/MeteoriteListItem";
export default {
  components: { MeteoriteListItem },
  data: () => ({
    meteorite: {
      name: "Aachen",
      id: "1",
      nametype: "Valid",
      recclass: "L5",
      mass: "21",
      fall: "Fell",
      year: "1880-01-01T00:00:00.000",
      reclat: "50.775000",
      reclong: "6.083330",
      geolocation: {
        type: "Point",
        coordinates: [6.08333, 50.775],
      },
    },
  }),
};
</script>

<style scoped lang="scss"></style>
