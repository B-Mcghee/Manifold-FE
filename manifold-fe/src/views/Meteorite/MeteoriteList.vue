<template>
  <v-data-iterator
    :items="meteorites"
    :items-per-page.sync="itemsPerPage"
    :page.sync="page"
    :search="search"
    :sort-by="sortBy.toLowerCase()"
    :sort-desc="sortDesc"
    hide-default-footer
    class="relative"
  >
    <!-- <template v-slot:header></template> -->
    <template v-slot:header>
      <v-toolbar dark color="#1b579e" class="mb-1 fixed" height="50px">
        <v-text-field
          v-model="search"
          clearable
          flat
          solo-inverted
          hide-details
          prepend-inner-icon="mdi-magnify"
          label="Search"
          class="mr-3 fixed"
          fixed
        >
        </v-text-field>
        <template v-if="$vuetify.breakpoint.smAndUp">
          <v-select
            v-model="sortBy"
            flat
            solo-inverted
            hide-details
            :items="keys"
            prepend-inner-icon="mdi-magnify"
            label="Sort by"
          >
          </v-select>
          <v-spacer></v-spacer>
          <v-btn-toggle v-model="sortDesc" mandatory>
            <v-btn large depressed color="#1b579e" :value="true">
              <v-icon>mdi-arrow-up</v-icon>
            </v-btn>
            <v-btn large depressed color="#1b579e" :value="false">
              <v-icon>mdi-arrow-down</v-icon></v-btn
            >
          </v-btn-toggle>
        </template>
      </v-toolbar>
    </template>

    <template v-slot:default="props">
      <div class="overflow">
        <v-row>
          <v-col
            align-center
            v-for="item in props.items"
            :key="item.id"
            cols="12"
            sm="6"
            md="3"
            lg="3"
          >
            <meteorite-list-item :meteorite="item"></meteorite-list-item>
          </v-col>
        </v-row>
      </div>
    </template>

    <template v-slot:footer>
      <v-row
        class="mt-2 footer blue--text text--darken-2 blue-grey lighten-5"
        align="center"
        justify="center"
      >
        <span class="">Items per page</span>
        <v-menu offset-y v-model="showCaret">
          <template v-slot:activator="{ on, attrs }">
            <v-btn
              dark
              text
              color="primary"
              class="ml-4"
              v-bind="attrs"
              v-on="on"
            >
              {{ itemsPerPage }}
              <v-icon>mdi-chevron-down</v-icon>
            </v-btn>
          </template>
          <v-list>
            <v-list-item
              v-for="(number, index) in itemsPerPageArray"
              :key="index"
              @click="updateItemsPerPage(number)"
            >
              <v-list-item-title>{{ number }}</v-list-item-title>
            </v-list-item>
          </v-list>
        </v-menu>

        <v-spacer></v-spacer>

        <span
          class="mr-3
            grey--text"
        >
          Page {{ page }} of {{ numberOfPages }}
        </span>
        <v-btn
          fab
          dark
          color="blue darken-3"
          class="mr-3"
          height="2em"
          width="2em"
          @click="formerPage"
        >
          <v-icon>mdi-chevron-left</v-icon>
        </v-btn>
        <v-btn
          fab
          dark
          color="blue darken-3"
          class=" mr-2"
          height="2em"
          width="2em"
          @click="nextPage"
        >
          <v-icon>mdi-chevron-right</v-icon>
        </v-btn>
      </v-row>
    </template>
  </v-data-iterator>
</template>

<script>
// import MeteoriteListItem from "../components/MeteoriteListItem.vue";
import MeteoriteListItem from "./MeteoriteListItem.vue";
export default {
  components: { MeteoriteListItem },
  data() {
    return {
      showCaret: false,
      softDesc: false,
      meteors: null,
      meteorClass: null,
      search1: "",
      itemsPerPageArray: [25, 50, 100],
      search: "",
      filter: {},
      sortDesc: false,
      page: 1,
      itemsPerPage: 25,
      sortBy: "name",
      keys: [
        "id",
        "name",
        "nametype",
        "recclass",
        "mass",
        "fall",
        "year",
        "reclat",
        "reclong",
        "geolocation",
      ],
    };
  },
  // components: { MeteoriteListItem },

  created() {
    this.meteors = JSON.parse(JSON.stringify(this.meteorites));
  },
  computed: {
    meteorites: function() {
      const array = this.$store.getters.initialMeteorites.map(
        (meteorite) => meteorite
      );

      return array.slice(0, 5);
    },
    numberOfPages() {
      return Math.ceil(this.meteorites.length / this.itemsPerPage);
    },
    filteredKeys() {
      return this.keys.filter((key) => key !== "Name");
    },
    filteredMeteors() {
      if (this.search1.length < 1)
        return this.$store.getters.initialMeteorites.map(
          (meteorite) => meteorite
        );

      return this.meteorites.filter((meteor) =>
        meteor.name.toLowerCase().includes(this.search1.toLowerCase())
      );
    },
  },
  methods: {
    nextPage() {
      if (this.page + 1 <= this.numberOfPages) this.page += 1;
    },
    formerPage() {
      if (this.page - 1 >= 1) this.page -= 1;
    },
    updateItemsPerPage(number) {
      this.itemsPerPage = number;
    },
    resort(recclass) {
      console.log("resort");
      this.meteors = JSON.parse(JSON.stringify(this.meteorites));
      this.meteorClass = this.meteors.filter(
        (meteor) => meteor.recclass == recclass
      );
    },
  },
};
</script>

<style>
.card {
  border: 3px solid red;
}
fixed {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
}

.overflow {
  height: 65vh;
  overflow: scroll;
}
.footer {
  padding: 4px 15px;
}
@media screen and (max-width: 1024px) {
  .overflow {
    height: 62vh;
    overflow: scroll;
  }
}
@media screen and (max-width: 1800px) {
  .overflow {
    height: 70vh;
    overflow: scroll;
  }
}
</style>
