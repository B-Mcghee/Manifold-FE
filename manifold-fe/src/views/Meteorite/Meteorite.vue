<template>
  <div class="relative">
    <div
      class="  d-flex justify-center text-h3 my-2 pb-4 font-weight-bold primary--text text-lg-h3"
    >
      Meteorites
    </div>
    <div>
      <v-container>
        <meteorite-list></meteorite-list>
      </v-container>

      <!-- <meteorite-list-item :meteorite="meteorite"></meteorite-list-item> -->
      <!-- <meteorite-list></meteorite-list> -->
    </div>
  </div>
</template>

<script>
import MeteoriteList from "./MeteoriteList.vue";

export default {
  components: { MeteoriteList },
  data: () => ({}),
  methods: {
    onStorageUpdate(event) {
      if (event.key === "favoriteMeteorites") {
        this.$store.dispatch("syncFavorites", event.newValue);
        this.name = event.newValue;
      }
    },
  },
  mounted() {
    if (localStorage.favoriteMeteorites) {
      this.name = localStorage.favoriteMeteorites;
    }
    window.addEventListener("storage", this.onStorageUpdate);
  },
  beforeDestroy() {
    window.removeEventListener("storage", this.onStorageUpdate);
  },
  watch: {
    name(newName) {
      localStorage.favoriteMeteorites = newName;
    },
  },
};
</script>

<style scoped>
.relative {
  position: relative;
  top: 0;
  right: 0;
  left: 0;
  bottom: 0;
}
</style>
