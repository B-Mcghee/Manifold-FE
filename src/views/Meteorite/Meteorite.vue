<template>
  <div class="relative">
    <div>
      <v-container fluid>
        <meteorite-list :category="categoryRoute"></meteorite-list>
      </v-container>
    </div>
  </div>
</template>

<script>
import { mapGetters } from "vuex";

import MeteoriteList from "./MeteoriteList.vue";
export default {
  components: { MeteoriteList },
  data() {
    return {
      category: undefined
    };
  },
  methods: {
    onStorageUpdate(event) {
      if (event.key === "favoriteMeteorites") {
        this.$store.dispatch("syncFavorites", event.newValue);
        this.name = event.newValue;
      }
    }
  },
  mounted() {
    this.category = "Meteorites";
    if (localStorage.favoriteMeteorites) {
      this.name = localStorage.favoriteMeteorites;
    }
    window.addEventListener("storage", this.onStorageUpdate);
  },
  beforeDestroy() {
    window.removeEventListener("storage", this.onStorageUpdate);
  },
  watch: {
    name(newName) {
      localStorage.favoriteMeteorites = newName;
    }
  },
  computed: {
    ...mapGetters(["favorites", "initialMeteorites", "meteoriteComparison"]),
    categoryRoute: function() {
      return this.$route.params;
    }
  }
};
</script>

<style scoped>
.relative {
  position: relative;
  top: 0;
  right: 0;
  left: 0;
  bottom: 0;
}
</style>
