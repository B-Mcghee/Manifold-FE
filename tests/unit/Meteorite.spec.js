import { createLocalVue, shallowMount, mount } from "@vue/test-utils";
import Vuetify from "vuetify";
import VueRouter from 'vue-router'
import Vuex from 'vuex'
import Meteorite from '@/views/Meteorite/Meteorite.vue'

const localVue = createLocalVue();
localVue.use(Vuex);
localVue.use(VueRouter);

describe("MeteoriteList.vue",  () => {

  let wrapper;
  let vuetify;
  let store;
let getters;
let state;
let router;
let category;

  beforeEach(() => {
    category = {property:'recclass', value: 'L5'}
      getters = {
            favorites: (state) => state.favoriteMeteorites,
            initialMeteorites: (state) => state.meteorites,
         meteoriteComparison: (state) => state.meteoriteComparison
      }
      state = {
          meteorites:[
            {
                name: "Aachen",
                id: "1",
                nametype: "Valid",
                recclass: "L5",
                mass: "21",
                fall: "Fell",
                year: "1880-01-01T00:00:00.000",
                reclat: "50.775000",
                reclong: "6.083330",
                geolocation: {
                  type: "Point",
                  coordinates: [6.08333, 50.775]
                }
              },
              {
                name: "Abee",
                id: "6",
                nametype: "Valid",
                recclass: "EH4",
                mass: "107000",
                fall: "Fell",
                year: "1952-01-01T00:00:00.000",
                reclat: "54.216670",
                reclong: "-113.000000",
                geolocation: {
                  Type: "Point",
                  coordinates: [
                    -113,
                    54.21667
                  ]
                }
              }
          ],
          favoriteMeteorites:[1],
          meteoriteComparison:[2,3], 

      }
      router = new VueRouter()
    store = new Vuex.Store({
        getters,
        state
    })
    vuetify = new Vuetify();
  });

  test('rending component', () => {
      const wrapper = shallowMount(Meteorite,{
          localVue, 
          vuetify, 
          store,
          router
      })

      expect(wrapper.html()).toMatchSnapshot(); 
  })
})