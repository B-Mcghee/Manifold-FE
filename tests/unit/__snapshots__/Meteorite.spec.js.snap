// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MeteoriteList.vue rending component 1`] = `
<div class="relative">
  <div>
    <v-container-stub tag="div" fluid="true">
      <meteorite-list-stub category="[object Object]"></meteorite-list-stub>
    </v-container-stub>
  </div>
</div>
`;
